fastcrypto-zkp/src/circom.rs

```diff
@@ -3,20 +3,100 @@
 
 use ark_bn254::{Fq, Fq2, G1Affine, G1Projective, G2Affine, G2Projective};
 use fastcrypto::error::FastCryptoError;
-
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
 /// A G1 point in BN254 serialized as a vector of three strings which is the canonical decimal
 /// representation of the projective coordinates in Fq.
-pub type CircomG1 = Vec<String>;
+pub type CircomG1 = Vec<Bn254Point>;
 
 /// A G2 point in BN254 serialized as a vector of three vectors each being a vector of two strings
 /// which are the canonical decimal representation of the coefficients of the projective coordinates
 /// in Fq2.
-pub type CircomG2 = Vec<Vec<String>>;
+pub type CircomG2 = Vec<Vec<Bn254Point>>;
+
+#[derive(Debug, Clone, JsonSchema, Eq, PartialEq)]
+pub struct Bn254Point([u8; 32]);
+
+impl Bn254Point {
+    pub fn unpadded(&self) -> &[u8] {
+        let mut buf = self.0.as_slice();
+
+        while !buf.is_empty() && buf[0] == 0 {
+            buf = &buf[1..];
+        }
+
+        // If the value is '0' then just return a slice of length 1 of the final byte
+        if buf.is_empty() {
+            &self.0[31..]
+        } else {
+            buf
+        }
+    }
+
+    pub fn padded(&self) -> &[u8] {
+        &self.0
+    }
+}
+impl std::str::FromStr for Bn254Point {
+    type Err = Bn254PointParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let big_int = <num_bigint::BigUint as num_traits::Num>::from_str_radix(s, 10)?;
+        let be_bytes = big_int.to_bytes_be();
+        let len = be_bytes.len();
+        let mut buf = [0; 32];
```

Comment:
Can we also ensure that `s` only contains the characters 0-9 in order to ensure that ser/de roundtrips